### ISSUE TITLE
Analyzer does not consistently process methods that return value types by reference

### ISSUE THREAD

**Issue by PathogenDavid** [szehetner/InliningAnalyzer#9]
The analyzer seems to have some issues with methods that return value types by reference.

Consider this code below:

```csharp
using System;

namespace InliningAnalyzerIgnoresRefReturnMethods
{
    public class Program
    {
        private static int a;
        private static int b;

        private static ref int Select(bool which)
            => ref (which ? ref a : ref b);

        private static int Select2(bool which)
            => Select(which);

        public static void Main(string[] args)
        {
            ref int x = ref Select(true);
            int y = Select(true);
            int z = Select2(true); // <-- Commenting out this line causes the call to Select by Select2 to be highlighted.
            Console.WriteLine(x);
            Console.WriteLine(y);
        }
    }
}
```

If you run the analyzer, none of the calls to `Select` are highlighted:

![image](https://user-images.githubusercontent.com/278957/47539170-9d436800-d894-11e8-8faf-93358f21e1e3.png)

Oddly enough, if you comment out the call to `Select2`, its call to `Select` in it is highlighted:

![image](https://user-images.githubusercontent.com/278957/47539210-ca901600-d894-11e8-8b88-95a1f9b3ed7b.png)

Even more strange, turning off "Prefer 32-bit" or explicitly choosing x64 will also fix `Select2`'s call to `Select`.

**Comment by szehetner** [comment]
There was a bug with matching the signature of ref return methods. This is fixed now, the 2 Select calls in Main are now also highlighted correctly.

The reason why it didn't work for x86 was that the Jit Compiler seems to emit the ETW event for the Select2->Select call twice, whereas it is only emitted once for x64. No idea why that is the case, but the doubled method call led to the overload resolution logic to kick in, which didn't work because of the missing ref return support. In the x64 case there is only one call, so the signature isn't needed to match the method call.

**Comment by PathogenDavid** [comment]
Thanks for the prompt fix!

**Comment by szehetner** [comment]
FYI, I just published a new version (0.7.3) that contains this fix.
